### ISSUE TITLE
[DOC] Tutorial: wrong order of arguments in custom objective function?

### ISSUE THREAD

**Issue by ingo-m** [dmlc/xgboost#5372]
In the tutorial on [Custom Objective and Evaluation Metric](https://xgboost.readthedocs.io/en/latest/tutorials/custom_metric_obj.html#customized-objective-function), the following custom objective function is provided as an example:

```python
def squared_log(predt: np.ndarray,
                dtrain: xgb.DMatrix) -> Tuple[np.ndarray, np.ndarray]:
    '''Squared Log Error objective. A simplified version for RMSLE used as
    objective function.
    '''
```

However, in the [documentation](https://xgboost.readthedocs.io/en/latest/python/python_api.html#module-xgboost.sklearn), it reads:

> A custom objective function can be provided for the `objective` parameter. In this case, it should have the signature `objective(y_true, y_pred) -> grad, hess`:

This makes me think that in the tutorial, the example objective function should be:

```python
def squared_log(dtrain: xgb.DMatrix,
                predt: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
    '''Squared Log Error objective. A simplified version for RMSLE used as
    objective function.
    '''
```

Should I make a PR to adjust the order or arguments in the tutorial, or did I get confused and there's actually no mistake?

**Comment by trivialfis** [comment]
@ingo-m  Thanks, but both the doc and example are correct.  The demo is written with native interface while the doc you quote is using Scikit-Learn wrapper.  Yeah consistency ... I know ...  Marked as a to-do item in https://github.com/dmlc/xgboost/issues/5152 .

**Comment by ingo-m** [comment]
Ok, sorry for the false alarm & thanks for clarifying @trivialfis .
